### PR TITLE
Deprecate log_names setting from filter processor

### PR DESCRIPTION
**Pull request by tigrannajaryan**
The LogRecord Name field [is removed](https://github.com/open-telemetry/opentelemetry-specification/pull/2271) from the specification and is [going to be removed](https://github.com/open-telemetry/opentelemetry-proto/pull/357) from the OTLP.

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Please add a note to the changelog to call out the deprecation 👍

**Comment by tigrannajaryan** [comment]
> Please add a note to the changelog to call out the deprecation 👍

Done.

File: internal/coreinternal/processor/filterconfig/config.go [hunk 1]
```diff
@@ -128,8 +129,12 @@ func (mp *MatchProperties) ValidateForLogs() error {
 		return errors.New("neither services nor span_names should be specified for log records")
 	}
 
-	if len(mp.LogNames) == 0 && len(mp.Attributes) == 0 && len(mp.Libraries) == 0 && len(mp.Resources) == 0 {
-		return errors.New(`at least one of "log_names", "attributes", "libraries" or "resources" field must be specified`)
+	if len(mp.LogNames) > 0 {
+		return errors.New("log_names setting is deprecated and will be removed soon")
```
Comment (tigrannajaryan) [inline internal/coreinternal/processor/filterconfig/config.go:133]: This is not what I intended to do. I think this needs to be a warning. Don't merge.

Comment (tigrannajaryan) [reply internal/coreinternal/processor/filterconfig/config.go:133]: The logger is not available here. It used to be available in component creation but I cannot see it anymore. Did we break it?

**Review by codeboten — APPROVED** [review]
👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +40,7 @@
 
 ## Deprecations
```
Comment (jpkrohling) [inline CHANGELOG.md:41]: Did we have this section in previous releases? If not, we need an emoji, to be in line with the other sections. How about one of those?

🗑️
🚩
👎

Comment (tigrannajaryan) [reply CHANGELOG.md]: We didn't. I like the flag one. Added.

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Looks like this change might have unintended consequences:

```
    filterlog_test.go:173: 
        	Error Trace:	filterlog_test.go:173
        	Error:      	Expected nil, but got: &errors.errorString{s:"at least one of \"attributes\", \"libraries\" or \"resources\" field must be specified"}
        	Test:       	TestLogRecord_Matching_True/log_name_match
    filterlog_test.go:174: 
        	Error Trace:	filterlog_test.go:174
        	Error:      	Expected value not to be nil.
        	Test:       	TestLogRecord_Matching_True/log_name_match
--- FAIL: TestLogRecord_Matching_True (0.00s)
    --- FAIL: TestLogRecord_Matching_True/log_name_match (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x87ff4a]

goroutine 12 [running]:
testing.tRunner.func1.2({0x8b7980, 0xbee400})
	/opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1209 +0x36c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1212 +0x3b6
panic({0x8b7980, 0xbee400})
	/opt/hostedtoolcache/go/1.17.6/x64/src/runtime/panic.go:1047 +0x266
github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterlog.TestLogRecord_Matching_True.func1(0x0)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/coreinternal/processor/filterlog/filterlog_test.go:177 +0x10a
testing.tRunner(0xc0001ae4e0, 0xc00000e2a0)
	/opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.17.6/x64/src/testing/testing.go:1306 +0x727
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterlog	0.031s
```

Is this an indication of a breaking change?

**Comment by tigrannajaryan** [comment]
> Is this an indication of a breaking change?

Yes, expected breaking change (LogNames is no longer one of the expected filters). The tests need to be fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,9 +40,10 @@
 - `resourcetotelemetry`: Ensure resource attributes are added to summary
   and exponential histogram data points. (#7523)
 
-## Deprecations
+## 🚩 Deprecations 🚩
 
 - Deprecated otel_to_hec_fields.name setting from splunkhec exporter. (#7560)
+- Deprecated log_names setting from filter processor. (#7552)
```
Comment (jpkrohling) [inline CHANGELOG.md:46]: I'm blocking this for a few moments, as this is now outdated due to the new release.

**Review by jpkrohling — DISMISSED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (tigrannajaryan) [reply CHANGELOG.md:62]: Updated now. Rebased from main.

**Comment by tigrannajaryan** [comment]
Discovered 2 bugs while updating the tests:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7755
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7756

**Pull request merged**